packages/mobile/src/qrcode/utils.ts

```diff
@@ -98,8 +99,14 @@ export function* handleBarcode(
   e164NumberToAddress: E164NumberToAddressType,
   secureSendTxData?: TransactionDataInput,
   isOutgoingPaymentRequest?: true,
-  requesterAddress?: string
+  requesterAddress?: string,
+  isWalletConnectRequest?: true
 ) {
+  if (isWalletConnectRequest) {
+    yield initialiseWalletConnect(barcode.data)
```

Comment:
Consider making this a `yield call(...` instead, it's more saga-esque 
Should work the same, just has some very subtle benefits.